test/WebSites/ValidationWebSite/Controllers/ModelMetadataTypeAttributeController.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.Mvc;
+using ValidationWebSite.ViewModels;
+
+namespace ValidationWebSite.Controllers
+{
+    public class ModelMetadataTypeValidationController : Controller
+    {
+        [HttpPost]
+        public object ValidateProductViewModelIncludingMetadata([FromBody] ProductViewModel product)
+        {
+            return CreateValidationDictionary();
+        }
+
+        [HttpPost]
+        public object ValidateSoftwareViewModelIncludingMetadata([FromBody] SoftwareViewModel software)
+        {
+            return CreateValidationDictionary();
+        }
+
+        [HttpPost]
+        public object TryValidateModelProductViewModelWithErrorInParameter(int id, [FromBody] ProductViewModel product)
+        {
+            //Clear ModelState entry if id is 0. TryValidateModel should not add entries that are cleared.
+            if (id == 0)
+            {
+                ModelState["id"].Errors.Clear();
+            }
+            TryValidateModel(product);
+
+            return CreateValidationDictionary();
+        }
+
+        [HttpGet]
+        public object TryValidateModelSoftwareViewModelNoPrefix()
```

Comment:
action name is no longer correct; should be `TryValidateModelSoftwareViewModelWithPrefix`